content/teknologi/altinnstudio/altinn-api/_index.md

```diff
@@ -7,1012 +7,61 @@ aliases:
-
-### Create and upload data
+Metadata for an App is the second job of the App API. There are endpoint that gives access to the metadata of the App itself, its data types and process deescription.
 
-Post a data file (e.g. an xml-document) as body (stream content) of request with matching headers. Must specify dataType as defined in the application metadata.
+Every App will expose almost identical endpoints and functionality. External parties should need only one client implementation across all App APIs. Technically there is nothing preventing an application owner from adding or making changes to the API, but in those cases it is probably an application with a different process flow. Applications like that would in most cases require some special handling and have their own documentation provided by the application owner.
 
 ```http
-POST {appPath}/instances/347829/41e57962-dfb7-4502-a4dd-8da28b0885fc/data?dataType=default
-```
-
-The request should have a content type and a content disposition which holds the data file's filename. You may also use a multipart, but  only the first part will be read.
-
-```http
-Content-Type = "image/png"
-Content-Disposition = attachment; filename=myfile.png
-
-PUT {appPath}/instances/347829/762011d1-d341-4c0a-8641-d8a104e83d30/data/692ee7df-82a9-4bba-b2f2-c8c4dac69aff
+https://{org}.apps.altinn.no/{org}/{appname}
 ```
 
-This call returns the data element metadata object that were created.
-
-```json
-{
-    "id": "692ee7df-82a9-4bba-b2f2-c8c4dac69aff",
-    "instanceGuid": "762011d1-d341-4c0a-8641-d8a104e83d30",
-    "dataType": "default",
-    "contentType": "application/xml",
-    "blobStoragePath": "org/app/762011d1-d341-4c0a-8641-d8a104e83d30/data/692ee7df-82a9-4bba-b2f2-c8c4dac69aff",
-    "selfLinks": {
-        "apps":   "{appPath}/instances/347829/762011d1-d341-4c0a-8641-d8a104e83d30/data/692ee7df-82a9-4bba-b2f2-c8c4dac69aff",
-        "platform": "{storagePath}/instances/347829/762011d1-d341-4c0a-8641-d8a104e83d30/data/692ee7df-82a9-4bba-b2f2-c8c4dac69aff"
-    },
-    "filename": "default.xml",
-    "created": "2019-03-06T15:00:23Z",
-    "createdBy": "org23",
-    "lastChanged": "2019-03-07T15:00:23Z",
-    "lastChangedBy": "org23",
-    "size": 20001,
-    "locked": false
-}
-```
-
-### Update data file
-
-Update (replace) a data element with a new one (payload). Data as multipart or as single body. Client does a PUT request to the App.
-It first calculates the data and replaces the existing data element. It returns the instance metadata to the client.
-
-
-![Flow chart for saving data](save-data.png "Save data")
-
-```http
-Content-Type = "image/png"
-Content-Disposition = attachment; filename=myfile.png
-
-PUT {appPath}/instances/347829/762011d1-d341-4c0a-8641-d8a104e83d30/data/692ee7df-82a9-4bba-b2f2-c8c4dac69aff
-```
+The URL identifies the application owner **org** specific hostname and the identificator of the App **org/appname**. 
```

Comment:
could we add some punctuation to this sentence to make it more readable? Thinking mostly about the "the application owner org specific hostname" part